### DISCUSSION TITLE
platform api: invalid token

### DISCUSSION THREAD

**Discussion by torquan** [chatwoot/chatwoot#3252] [Self hosted] [answered]
I'm selfhosting chatwoot (via caprover). I created a PlattformApp and obtained the token:

```
app = PlatformApp.create(name: {my app name})
app.access_token.token
```

However, when I send for example a POST request to `https://{host}/platform/api/v1/accounts`, the response is 
```
{
    "error": "Invalid access_token"
}
```

I verified that the token is correct (by calling `PlatformApp.last.access_token`). The token is a header param ("api_access_token") as described in the API doc.

Any tips on why I receive this error?

Update:
1) tried creating a second platform app. That key doesn't work either.
2) Here's the error coming from: [platform_controller.rb](https://github.com/chatwoot/chatwoot/blob/develop/app/controllers/platform_controller.rb)
3) I have the same problem trying to authenicate with userApiKey. I've been sending the requests via axios and Postman.

**Comment by sojan-official** [comment, marked as answer]
if your installation is behind an Nginx frontend proxy, you will want to add the following to the Nginx headers for the auth headers to be passed into chatwoot 
https://www.chatwoot.com/docs/self-hosted/deployment/caprover#common-errors

**Reply by torquan** [reply]
Thank you very much.
